### PR TITLE
Amazon Linux 2017 support.

### PR DESCRIPTION
**Pull request by o0oxid**
Amazon has recently released AMI 2017.03. Fact `:operatingsystemmajrelease` is now 2017. So this PR adds latest AMI support.

### CODE REVIEW

**Comment by ghoneycutt** [comment]
Thanks for the contribution! Looks like I need to get PR #50 merged and then have you rebase against it so that the tests will pass.

File: spec/classes/init_spec.rb [hunk 1]
```diff
@@ -1,10 +1,7 @@
 require 'spec_helper'
 describe 'nscd' do
-  platforms = {
-    'debian6' =>
-      {
-        :osfamily                  => 'Debian',
-        :operatingsystemmajrelease => '6',
+  defaults = {
```
Comment (ghoneycutt) [inline spec/classes/init_spec.rb]: Love this!

File: manifests/init.pp [hunk 2]
```diff
@@ -363,7 +363,7 @@
         # Added for Amazon Linux support
         # Amazon Linux has 'YYYY-MM' version format
         # https://github.com/ghoneycutt/puppet-module-nscd/issues/41
-        '2015', '2016': {
+        /^201[5-7]+$/: {
```
Comment (ghoneycutt) [inline manifests/init.pp]: Could you please add 2017 to the README under Compatibility.

https://github.com/ghoneycutt/puppet-module-nscd#compatibility

Comment (o0oxid) [inline manifests/init.pp]: Done, thanks for review.

**Comment by ghoneycutt** [comment]
Thanks again @o0oxid 

I have rebased this commit in PR #52

**Pull request closed without merging**